tools/shellcheck-gitrange.bash

```diff
@@ -6,16 +6,22 @@ set -e
 
 # Sample usage:
 #
-#   $  shellcheck-gitrange.bash HEAD~5.. [ -f gcc ]
+#   $  shellcheck-gitrange.bash origin/master... [ -f gcc ]
 
 main()
 {
     # The rest of args is passed as is to shellcheck
-    local gitrange="$1"; shift
+    local diffrange="$1"; shift
 
-    printf '%s checking range %s\n\n' "$0" "$gitrange"
-    # also a sanity check
-    git log --oneline "$gitrange" -- | cat # no pager
+    printf '%s checking diff range: %s\n\n' "$0" "$diffrange"
+
+    # Triple dot "git log A...B" includes commits not relevant to triple
+    # dot "git diff A...B"
+    local logrange=${diffrange/.../..}
+    ( set -x
+      # also a sanity check of the argument
+      git --no-pager log --oneline --graph --stat --max-count=40 "$logrange" --
```

Comment:
@marc-hb I'm not very clear on this part. We are using git log to check whether the argument $1 is set right or not? If so, why don't we use `git diff` directly here? This may be a little same with @ranj063's question.

Reply:
This git log command shows which commits are being checked for logging purposes. It also aborts the script early if "$1" is not valid.

The git diff command below extracts all the names of the files (`fname`) modified by the PR

Those are two different operations that need different commands with a slightly different argument. I'm not sure I understand the question.

Reply:
@libinyang I believe git log is still the correct command for doing the sanity check, so this seems correct.